### DISCUSSION TITLE
feature: how to handle more than 1 repository at the same time (self hosted)

### DISCUSSION THREAD

**Discussion by Jiehong** [renovatebot/renovate#8999] [Request Help]
We're running renovate as a self-hosted app (docker image running as a kubernetes cronjob), and it's set to all repos of a given bitbucket project.

However, it does check each repository one by one, and so it takes quite a while (like 1 hour for 50 repositories)

Is there a way to make renovate process like 5 repositories in parallel?

I was unable to find an option for that.

**Comment by viceice** [comment]
There is not parallel repo processing. You can try to use autodiscover filter and multiple jobs to spread processing in parallel.

Or use the free renovate on-prem version, it has state full queue and web hooks Reaktion

**Reply by Jiehong** [reply]
Ah, I see. Couldn't my ticket as a feature request be valid then?

**Reply by viceice** [reply]
Currently it's not possible to implement parallel repo processing. There are to many conditions which are blocking this

**Reply by timmkrause** [reply]
@Jiehong How did you achieve the multi-repo approach? Are you running Renovate in a loop manually passing each repo or is it capable to pass multiple repos in a list?

@viceice I assume multi-repo is not possible using the [Renovate Me](https://marketplace.visualstudio.com/items?itemName=jyc.vsts-extensions-renovate-me) Azure DevOps task? So we would need to switch to either use the CLI directly or via the Docker image, is that correct?

**Reply by timmkrause** [reply]
Found [this one](https://docs.renovatebot.com/setup-azure-devops/#using-a-single-pipeline-to-update-multiple-repositories), sounds promising. Please ignore my comment.

**Reply by viceice** [reply]
@timmkrause `Renovate Me` is not an official task from renovate maintainer, you need to ask at https://github.com/jycouet/VSTSExtensions for help on it.

Otherwise you are welcome and it seems you found our docs 😉

**Reply by Jiehong** [reply]
@timmkrause : you can set a glob pattern for renovate: "autodiscoverFilter": "project/*",

Then renovate will handle all the repositories part of the "project" (in my case, a bitbucket project).
